test/e2e/olm_upgrade_test.go

```diff
@@ -0,0 +1,333 @@
+								VolumeSource: corev1.VolumeSource{
+									ConfigMap: &corev1.ConfigMapVolumeSource{
+										LocalObjectReference: corev1.LocalObjectReference{
+											Name: config.Name,
+										},
+									},
+								},
+							},
+						},
+						RestartPolicy:      corev1.RestartPolicyNever,
+						ServiceAccountName: serviceAccount.Name,
+					},
+				},
+			},
+		}
+		job, err = test.Client().Core().BatchV1().Jobs(namespace.Name).Create(test.Ctx(), job, metav1.CreateOptions{})
+		test.Expect(err).NotTo(HaveOccurred())
+		test.T().Logf("Created Job %s/%s successfully", job.Namespace, job.Name)
+
+		// Retrieving the job logs once it has completed or timed out
+		defer WriteJobLogs(test, job.Namespace, job.Name)
+
+		test.T().Logf("Waiting for Job %s/%s to complete", job.Namespace, job.Name)
+		test.Eventually(Job(test, job.Namespace, job.Name), TestTimeoutLong).Should(
+			Or(
+				WithTransform(ConditionStatus(batchv1.JobComplete), Equal(corev1.ConditionTrue)),
+				WithTransform(ConditionStatus(batchv1.JobFailed), Equal(corev1.ConditionTrue)),
+			))
+
+		// Assert the job has completed successfully
+		test.Expect(GetJob(test, job.Namespace, job.Name)).
+			To(WithTransform(ConditionStatus(batchv1.JobComplete), Equal(corev1.ConditionTrue)))
+
+	} else {
+		test.T().Skip("Skipping OLM upgarde test because RUN_OLM_TESTS is not set")
+	}
+}
+
+// Submit a Job to the Ray cluster and trains the MNIST dataset using the CodeFlare SDK.
+func TestMnistJobSubmit(t *testing.T) {
+
+	test := With(t)
+	test.T().Parallel()
+	if os.Getenv("RUN_OLM_TESTS") == "true" {
+
+		//delete the namespace after test complete
+		defer DeleteTestNamespace(test, namespaceName)
+
+		// Test configuration
+		config := &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: corev1.SchemeGroupVersion.String(),
+				Kind:       "ConfigMap",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "mnist-ray-job",
+				Namespace: namespaceName,
+			},
+			BinaryData: map[string][]byte{
+				// SDK script
+				"mnist_rayjob.py": ReadFile(test, "mnist_rayjob.py"),
+				// pip requirements
+				"requirements.txt": ReadFile(test, "mnist_pip_requirements.txt"),
+				// MNIST training script
+				"mnist.py": ReadFile(test, "mnist.py"),
+			},
+			Immutable: Ptr(true),
+		}
+		config, err := test.Client().Core().CoreV1().ConfigMaps(namespaceName).Create(test.Ctx(), config, metav1.CreateOptions{})
+		test.Expect(err).NotTo(HaveOccurred())
+		test.T().Logf("Created ConfigMap %s/%s successfully", config.Namespace, config.Name)
+
+		// SDK client RBAC
+		serviceAccount := &corev1.ServiceAccount{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: corev1.SchemeGroupVersion.String(),
+				Kind:       "ServiceAccount",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "ray-user",
+				Namespace: namespaceName,
+			},
+		}
+		serviceAccount, err = test.Client().Core().CoreV1().ServiceAccounts(namespaceName).Create(test.Ctx(), serviceAccount, metav1.CreateOptions{})
```

Comment:
This user is missing role to get Routes:
```

HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"routes.route.openshift.io is forbidden: User \"system:serviceaccount:test-ns-olmupgrade:ray-user\" cannot list resource \"routes\" in API group \"route.openshift.io\" in the namespace \"test-ns-olmupgrade\"","reason":"Forbidden","details":{"group":"route.openshift.io","kind":"routes"},"code":403}
```
You may reuse "sdk-user" from above for it.

Reply:
Done